src/types.js

```diff
@@ -40,3 +40,39 @@ export type ServerKeyRepository = {
   }>,
   getPrivateKey: () => Promise<?string>,
 };
+
+export type TokenObject = {
+  accessToken: string,
+  accessTokenExpiresAt: Date,
+  refreshToken: string,
+  refreshTokenExpiresAt: Date,
+  scope: string,
+};
+
+export type User = {
+  accessTokens: Array<TokenObject>,
+  claimCodes: Array<string>,
+  created_at: Date,
+  id: string,
+  passwordHash: string,
+  salt: string,
+  username: string,
+};
+
+export type UserCredentials = {
+  username: string,
+  password: string,
+};
+
+export type UserRepository = Repository<User> & {
```

Comment:
You should only need the functions that actually get called in `spark-protocol`. That way you can keep a smaller interface (type)